cpp/bench/ann/src/cuvs/cuvs_ann_bench_param_parser.h

```diff
@@ -176,7 +188,11 @@ nlohmann::json collect_conf_with_prefix(const nlohmann::json& conf,
 
 template <typename T, typename IdxT>
 void parse_build_param(const nlohmann::json& conf,
+#if defined(CUVS_ANN_BENCH_USE_CUVS_MG)
+                       typename cuvs::bench::cuvs_mg_cagra<T, IdxT>::build_param& param)
```

Comment:
This function parses the single GPU parameters and puts them either in a single GPU or multiple GPU bench struct. This function should indeed work for both cases. [The MG benchmarks CUDA files are parameterized with `CUVS_ANN_BENCH_USE_CUVS_MG`](https://github.com/rapidsai/cuvs/blob/bcf97c9e8f35f8d8e7830ae8ac8f3cff99532d56/cpp/bench/ann/CMakeLists.txt#L259-L261).

Reply:
Thanks for the clarification! Now I see why this can work.

I still think this is an unfortunate design. This is added complexity that is not necessary. The function here reads only `cuvs_cagra<T, IdxT::build_param` and not `cuvs_mg_cagra<T, IdxT>::build_param`. We could just leave it as is with `cuvs_cagra<T, IdxT::build_param`.

Reply:
Could fix this with my latest commit.